### ISSUE TITLE
[1.13.2/1.12.2] Crashing on joining specific server

### ISSUE THREAD

**Issue by j13li78** [ImpactDevelopment/ImpactClient#1241]
**Describe the bug**
When i join a server called twerion.net i crash after seeing my character.

**To Reproduce**
1. Join Server: twerion.net
2. Wait for your Character to be visible
3. Crash

**Expected behavior**
Not crashing.

<details> 
  <summary><strong>Game output</strong></summary>
 

```
The game crashed whilst unexpected error
Error: java.lang.ClassCastException: java.lang.Byte cannot be cast to java.lang.Float
```

</details>

<details> 
  <summary><strong>Crash report</strong></summary>
 

```
---- Minecraft Crash Report ----
// I feel sad now :(

Time: 31.05.19 17:36
Description: Unexpected error

java.lang.ClassCastException: java.lang.Byte cannot be cast to java.lang.Float
	at aog.cQ(SourceFile:2069)
	at me.zero.clarinet.3v.lambda$null$7(3v.java:99)
	at me.zero.clarinet.3v$$Lambda$2771/676875212.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at me.zero.clarinet.3v.lambda$new$8(3v.java:177)
	at me.zero.clarinet.3v$$Lambda$1597/1990666779.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.bus.EventManager.lambda$post$5(EventManager.java:99)
	at me.zero.alpine.bus.EventManager$$Lambda$1133/2099091301.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.bus.EventManager.post(EventManager.java:99)
	at clientapi.7.post(7.java:7)
	at cty.handler$updateCameraAndRender$zzh000(EntityRenderer.java:2393)
	at cty.a(EntityRenderer.java:991)
	at cft.c(SourceFile:850)
	at cft.a(SourceFile:395)
	at net.minecraft.client.main.Main.main(SourceFile:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at aog.cQ(SourceFile:2069)
	at me.zero.clarinet.3v.lambda$null$7(3v.java:99)
	at me.zero.clarinet.3v$$Lambda$2771/676875212.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at me.zero.clarinet.3v.lambda$new$8(3v.java:177)
	at me.zero.clarinet.3v$$Lambda$1597/1990666779.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.bus.EventManager.lambda$post$5(EventManager.java:99)
	at me.zero.alpine.bus.EventManager$$Lambda$1133/2099091301.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.bus.EventManager.post(EventManager.java:99)
	at clientapi.7.post(7.java:7)
	at cty.handler$updateCameraAndRender$zzh000(EntityRenderer.java:2393)

-- Affected level --
Details:
	Level name: MpServer
	All players: 21 total; [ctj['DoctorKlenk'/11687765, l='MpServer', x=-35.50, y=37.00, z=-287.50], ctk['PlayriGamer'/11597960, l='MpServer', x=-32.81, y=37.00, z=-287.50], ctk['Der1Duke'/11675148, l='MpServer', x=-35.50, y=95.00, z=-287.50], ctk['Edohand'/11637857, l='MpServer', x=-35.50, y=37.00, z=-287.50], ctk['CuzImNooby'/11667979, l='MpServer', x=-31.41, y=37.00, z=-287.50], ctk['Enapay'/11682381, l='MpServer', x=-27.16, y=69.00, z=-274.59], ctk['DayMonGa'/11677943, l='MpServer', x=-11.16, y=38.97, z=-277.63], ctk['§U§d§R§C§1§n§5§i'/10000013, l='MpServer', x=-35.50, y=38.00, z=-295.50], ctk['§G§8§6§A§6§a§8§x'/10000018, l='MpServer', x=-32.50, y=38.00, z=-279.50], ctk['§a§q§T§W§R§j§r§j'/10000022, l='MpServer', x=-35.50, y=38.00, z=-279.50], ctk['§7§A§8§Q§r§h§m§y'/10000028, l='MpServer', x=-38.50, y=38.00, z=-295.50], ctk['§b§8§8§1§0§f§b§L'/10000032, l='MpServer', x=-5.50, y=37.00, z=-280.50], ctk['§H§1§g§l§J§P§4§L'/10000037, l='MpServer', x=-3.50, y=37.00, z=-292.50], ctk['§y§R§J§i§D§k§Q§s'/10000042, l='MpServer', x=-3.50, y=37.00, z=-287.50], ctk['§e§Q§0§D§9§q§x§x'/10000047, l='MpServer', x=-5.50, y=37.00, z=-294.50], ctk['§p§3§n§H§9§I§8§i'/10000052, l='MpServer', x=-9.50, y=37.00, z=-298.50], ctk['§I§X§N§S§O§G§m§c'/10000057, l='MpServer', x=-9.50, y=37.00, z=-276.50], ctk['§q§Y§X§G§G§Q§R§k'/10000062, l='MpServer', x=-7.50, y=37.00, z=-296.50], ctk['§b§N§j§l§B§2§L§n'/10000067, l='MpServer', x=-3.50, y=37.00, z=-282.50], ctk['§h§q§m§m§U§E§C§n'/10000072, l='MpServer', x=-7.50, y=37.00, z=-278.50], ctk['§R§P§9§e§O§C§z§y'/10000009, l='MpServer', x=-38.50, y=38.00, z=-279.50]]
	Chunk stats: MultiplayerChunkCache: 220, 220
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (0,64,0), Chunk: (at 0,4,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 2692796 game time, 1756500 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: adventure (ID 2). Hardcore: false. Cheats: false
	Forced entities: 101 total; [ctj['DoctorKlenk'/11687765, l='MpServer', x=-35.50, y=37.00, z=-287.50], alz[''/10000024, l='MpServer', x=-35.50, y=38.69, z=-279.50], alz[''/10000025, l='MpServer', x=-35.50, y=38.38, z=-279.50], alz[''/10000026, l='MpServer', x=-35.50, y=38.09, z=-279.50], alz[''/10000027, l='MpServer', x=-35.50, y=37.78, z=-279.50], ctk['§7§A§8§Q§r§h§m§y'/10000028, l='MpServer', x=-38.50, y=38.00, z=-295.50], alz[''/10000029, l='MpServer', x=-38.50, y=39.00, z=-295.50], alz[''/10000030, l='MpServer', x=-38.50, y=38.69, z=-295.50], alz[''/10000031, l='MpServer', x=-38.50, y=38.38, z=-295.50], alz[''/10000016, l='MpServer', x=-35.50, y=38.38, z=-295.50], alz[''/10000017, l='MpServer', x=-35.50, y=38.09, z=-295.50], ctk['§G§8§6§A§6§a§8§x'/10000018, l='MpServer', x=-32.50, y=38.00, z=-279.50], alz[''/10000019, l='MpServer', x=-32.50, y=39.00, z=-279.50], alz[''/10000020, l='MpServer', x=-32.50, y=38.69, z=-279.50], alz[''/10000021, l='MpServer', x=-32.50, y=38.38, z=-279.50], ctk['§a§q§T§W§R§j§r§j'/10000022, l='MpServer', x=-35.50, y=38.00, z=-279.50], alz[''/10000023, l='MpServer', x=-35.50, y=39.00, z=-279.50], ctk['§R§P§9§e§O§C§z§y'/10000009, l='MpServer', x=-38.50, y=38.00, z=-279.50], alz[''/10000010, l='MpServer', x=-38.50, y=39.00, z=-279.50], alz[''/10000011, l='MpServer', x=-38.50, y=38.69, z=-279.50], alz[''/10000012, l='MpServer', x=-38.50, y=38.38, z=-279.50], ctk['§U§d§R§C§1§n§5§i'/10000013, l='MpServer', x=-35.50, y=38.00, z=-295.50], alz[''/10000014, l='MpServer', x=-35.50, y=39.00, z=-295.50], alz[''/10000015, l='MpServer', x=-35.50, y=38.69, z=-295.50], ano['Schleim'/8222053, l='MpServer', x=-37.59, y=38.88, z=-286.41], alz['Rüstungsständer'/8222052, l='MpServer', x=-37.59, y=38.88, z=-286.41], alz['Offline'/10000056, l='MpServer', x=-9.50, y=36.69, z=-298.50], ctk['§I§X§N§S§O§G§m§c'/10000057, l='MpServer', x=-9.50, y=37.00, z=-276.50], alz['Rüstungsständer'/10000058, l='MpServer', x=-9.50, y=38.00, z=-276.50], amm['Sandstein'/10000059, l='MpServer', x=-9.50, y=39.48, z=-276.50], alz['Fastbuilder'/10000060, l='MpServer', x=-9.50, y=37.00, z=-276.50], alz[''/10000061, l='MpServer', x=-9.50, y=36.69, z=-276.50], ctk['§q§Y§X§G§G§Q§R§k'/10000062, l='MpServer', x=-7.50, y=37.00, z=-296.50], alz['Rüstungsständer'/10000063, l='MpServer', x=-7.50, y=38.00, z=-296.50], alz['Rüstungsständer'/10000048, l='MpServer', x=-5.50, y=38.00, z=-294.50], amm['Diamantspitzhacke'/10000049, l='MpServer', x=-5.50, y=39.48, z=-294.50], alz['Creative'/10000050, l='MpServer', x=-5.50, y=37.00, z=-294.50], alz[''/10000051, l='MpServer', x=-5.50, y=36.69, z=-294.50], ctk['§p§3§n§H§9§I§8§i'/10000052, l='MpServer', x=-9.50, y=37.00, z=-298.50], alz['Rüstungsständer'/10000053, l='MpServer', x=-9.50, y=38.00, z=-298.50], amm['Eisenschwert'/10000054, l='MpServer', x=-9.50, y=39.48, z=-298.50], alz[''/10000055, l='MpServer', x=-9.50, y=37.00, z=-298.50], alz['CityBuild'/10000040, l='MpServer', x=-3.50, y=37.00, z=-292.50], alz[''/10000041, l='MpServer', x=-3.50, y=36.69, z=-292.50], ctk['§y§R§J§i§D§k§Q§s'/10000042, l='MpServer', x=-3.50, y=37.00, z=-287.50], alz['Rüstungsständer'/10000043, l='MpServer', x=-3.50, y=38.00, z=-287.50], amm['Grasblock'/10000044, l='MpServer', x=-3.50, y=39.48, z=-287.50], alz['SkyBlock'/10000045, l='MpServer', x=-3.50, y=37.00, z=-287.50], alz[''/10000046, l='MpServer', x=-3.50, y=36.69, z=-287.50], ctk['§e§Q§0§D§9§q§x§x'/10000047, l='MpServer', x=-5.50, y=37.00, z=-294.50], ctk['PlayriGamer'/11597960, l='MpServer', x=-32.81, y=37.00, z=-287.50], ctk['§b§8§8§1§0§f§b§L'/10000032, l='MpServer', x=-5.50, y=37.00, z=-280.50], alz['Rüstungsständer'/10000033, l='MpServer', x=-5.50, y=38.00, z=-280.50], ctk['PlayriGamer'/11597960, l='MpServer', x=-32.81, y=37.00, z=-287.50], amm['Rotes Bett'/10000034, l='MpServer', x=-5.50, y=39.48, z=-280.50], ctk['Der1Duke'/11675148, l='MpServer', x=-35.50, y=95.00, z=-287.50], alz['BedWars'/10000035, l='MpServer', x=-5.50, y=37.00, z=-280.50], ctk['Edohand'/11637857, l='MpServer', x=-35.50, y=37.00, z=-287.50], alz[''/10000036, l='MpServer', x=-5.50, y=36.69, z=-280.50], ctk['CuzImNooby'/11667979, l='MpServer', x=-31.41, y=37.00, z=-287.50], ctk['§H§1§g§l§J§P§4§L'/10000037, l='MpServer', x=-3.50, y=37.00, z=-292.50], alz['Rüstungsständer'/10000038, l='MpServer', x=-3.50, y=38.00, z=-292.50], ctk['Enapay'/11682381, l='MpServer', x=-27.16, y=69.00, z=-274.59], amm['Eisenspitzhacke'/10000039, l='MpServer', x=-3.50, y=39.48, z=-292.50], ctk['FnaticXizt'/11687038, l='MpServer', x=-55.16, y=34.00, z=-263.63], ctk['DayMonGa'/11677943, l='MpServer', x=-11.16, y=38.97, z=-277.63], ctk['§U§d§R§C§1§n§5§i'/10000013, l='MpServer', x=-35.50, y=38.00, z=-295.50], ctk['§G§8§6§A§6§a§8§x'/10000018, l='MpServer', x=-32.50, y=38.00, z=-279.50], ctk['DayMonGa'/11677943, l='MpServer', x=-11.16, y=38.97, z=-277.63], ctk['§a§q§T§W§R§j§r§j'/10000022, l='MpServer', x=-35.50, y=38.00, z=-279.50], ctk['§7§A§8§Q§r§h§m§y'/10000028, l='MpServer', x=-38.50, y=38.00, z=-295.50], ctk['§b§8§8§1§0§f§b§L'/10000032, l='MpServer', x=-5.50, y=37.00, z=-280.50], ctk['§H§1§g§l§J§P§4§L'/10000037, l='MpServer', x=-3.50, y=37.00, z=-292.50], ctk['§y§R§J§i§D§k§Q§s'/10000042, l='MpServer', x=-3.50, y=37.00, z=-287.50], ctk['§e§Q§0§D§9§q§x§x'/10000047, l='MpServer', x=-5.50, y=37.00, z=-294.50], ctk['§p§3§n§H§9§I§8§i'/10000052, l='MpServer', x=-9.50, y=37.00, z=-298.50], ctk['§I§X§N§S§O§G§m§c'/10000057, l='MpServer', x=-9.50, y=37.00, z=-276.50], ctk['§q§Y§X§G§G§Q§R§k'/10000062, l='MpServer', x=-7.50, y=37.00, z=-296.50], ctk['§b§N§j§l§B§2§L§n'/10000067, l='MpServer', x=-3.50, y=37.00, z=-282.50], ctk['§h§q§m§m§U§E§C§n'/10000072, l='MpServer', x=-7.50, y=37.00, z=-278.50], ctk['§h§q§m§m§U§E§C§n'/10000072, l='MpServer', x=-7.50, y=37.00, z=-278.50], ctk['§R§P§9§e§O§C§z§y'/10000009, l='MpServer', x=-38.50, y=38.00, z=-279.50], alz['Rüstungsständer'/10000073, l='MpServer', x=-7.50, y=38.00, z=-278.50], amm['Eisengitter'/10000074, l='MpServer', x=-7.50, y=39.48, z=-278.50], alz[''/10000075, l='MpServer', x=-7.50, y=37.00, z=-278.50], alz['Offline'/10000076, l='MpServer', x=-7.50, y=36.69, z=-278.50], amm['Trichterlore'/10000064, l='MpServer', x=-7.50, y=39.48, z=-296.50], alz['GTA'/10000065, l='MpServer', x=-7.50, y=37.00, z=-296.50], alz[''/10000066, l='MpServer', x=-7.50, y=36.69, z=-296.50], ctk['§b§N§j§l§B§2§L§n'/10000067, l='MpServer', x=-3.50, y=37.00, z=-282.50], alz['Rüstungsständer'/10000068, l='MpServer', x=-3.50, y=38.00, z=-282.50], amm['Feder'/10000069, l='MpServer', x=-3.50, y=39.48, z=-282.50], alz['SkyPvP'/10000070, l='MpServer', x=-3.50, y=37.00, z=-282.50], alz[''/10000071, l='MpServer', x=-3.50, y=36.69, z=-282.50], ctk['CuzImNooby'/11667979, l='MpServer', x=-31.41, y=37.00, z=-287.50], ctk['Der1Duke'/11675148, l='MpServer', x=-35.50, y=95.00, z=-287.50], ann['Enapay's Tier'/11682683, l='MpServer', x=-26.31, y=69.00, z=-275.59], ctk['Edohand'/11637857, l='MpServer', x=-35.50, y=37.00, z=-287.50], alz['Rüstungsständer'/11668312, l='MpServer', x=-31.94, y=38.97, z=-288.38], ano['Schleim'/11668313, l='MpServer', x=-31.94, y=38.97, z=-288.38], ctk['Enapay'/11682381, l='MpServer', x=-27.16, y=69.00, z=-274.59]]
	Retry entities: 0 total; []
	Server brand: BungeeCord (VenomCord 1.14) <- Spigot
	Server type: Non-integrated multiplayer server
Stacktrace:
	at crg.a(WorldClient.java:576)
	at cft.b(SourceFile:1815)
	at cft.a(SourceFile:408)
	at net.minecraft.client.main.Main.main(SourceFile:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.13.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 389756448 bytes (371 MB) / 872415232 bytes (832 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Launched Version: 1.13.2-Impact_4.6
	LWJGL: 3.1.6 build 14
	OpenGL: GeForce GTX 1070 Ti/PCIe/SSE2 GL version 4.6.0 NVIDIA 419.17, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: Deutsch (Deutschland)
	Profiler Position: N/A (disabled)
	CPU: 8x Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
	OptiFine Version: OptiFine_1.13.2_HD_U_E5
	OptiFine Build: 20181221-162116
	Render Distance Chunks: 8
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 NVIDIA 419.17
	OpenGlRenderer: GeForce GTX 1070 Ti/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 8
```

</details>

<details> 
  <summary><strong>Additional Client Info</strong></summary>
 

- Impact Version: e.g. "4.6 for 1.12.2/1.13.2"
- Using Optifine? Yes 
- Using LiteLoader? No 
- Enabled mods:  "Velocity[0%,0%], ESP[Shader], Jesus[Solid], Log Position, Nametags, Anti Blind, No Slow, No SRP, Sprint, Light"
- Operating System:  "Windows 10 64bit"

</details>

**Comment by ngt1133** [comment]
same :/

**Comment by AstragoYT** [comment]
same

**Comment by ZeroMemes** [comment]
Disable Nametags for now, looking into a fix.

**Comment by 0-x-2-2** [comment]
😹

**Comment by leijurv** [comment]
```
java.lang.ClassCastException: java.lang.Byte cannot be cast to java.lang.Float
	at net.minecraft.entity.player.EntityPlayer.getAbsorptionAmount(EntityPlayer.java:2066)
```

```
   public float getAbsorptionAmount() {
      return this.getDataManager().get(ABSORPTION);
   }
```

wild

**Comment by ZeroMemes** [comment]
LMAO?

**Comment by leijurv** [comment]
yeah what the actual fuck

**Comment by leijurv** [comment]
My bet is that it's Entity.setFlag calling this.dataManager.set(FLAGS, (byte) 1<<flag) blah blah

**Comment by leijurv** [comment]
Before it crashes, it logs `Textures payload has been tampered with (signature invalid)`

**Comment by leijurv** [comment]
Looked deeper into it, and oh dear.

I printed out SPacketSpawnPlayer as it came in, and, well, Minecraft does no checking and it just accepts whatever types are in it. The DataEntry is for ABSORBTION but the type is Byte. 

`} catch(ClassCastException` time :sunglasses:

**Comment by leijurv** [comment]
One interesting thing is that this only happens once you move the mouse or WASD. If you're afk it never sends these packets.

**Comment by ZeroMemes** [comment]
We could intercept `SPacketEntityMetadata` to prevent the client from receiving any invalid data at all.

**Comment by leijurv** [comment]
We could, and I tried but holy shit what a pain in the ass lmao

**Comment by ZeroMemes** [comment]
Fixed in 4.7
